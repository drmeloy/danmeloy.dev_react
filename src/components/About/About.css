.About {
  padding-left: 6%;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  min-height: 100vh;
  color: white;

  p {
    font-family: 'Raleway', sans-serif;
    font-size: 1em;
    width: 50vw;
    line-height: 2;
    text-align: right;
  }

  @media (min-width: 10px) and (max-width: 720px){
    p {
      font-size: .7rem;
      line-height: 1.6;
    }
  }

  @media (min-width: 720px) and (max-width: 1023px){
    p {
      font-size: 1.3rem;
      line-height: 2;
    }
  }

  @media (max-width: 1025px) {
    padding-left: 0;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding-top: 1em;
    padding-bottom: 1em;

    p {
      width: 90vw;
      margin: 1% auto;
      text-align: center
    }
  }

  background: linear-gradient(216deg, rgba(77, 77, 77,0.05) 0%, rgba(77, 77, 77,0.05) 25%,rgba(42, 42, 42,0.05) 25%, rgba(42, 42, 42,0.05) 38%,rgba(223, 223, 223,0.05) 38%, rgba(223, 223, 223,0.05) 75%,rgba(36, 36, 36,0.05) 75%, rgba(36, 36, 36,0.05) 100%),linear-gradient(44deg, rgba(128, 128, 128,0.05) 0%, rgba(128, 128, 128,0.05) 34%,rgba(212, 212, 212,0.05) 34%, rgba(212, 212, 212,0.05) 57%,rgba(25, 25, 25,0.05) 57%, rgba(25, 25, 25,0.05) 89%,rgba(135, 135, 135,0.05) 89%, rgba(135, 135, 135,0.05) 100%),linear-gradient(241deg, rgba(55, 55, 55,0.05) 0%, rgba(55, 55, 55,0.05) 14%,rgba(209, 209, 209,0.05) 14%, rgba(209, 209, 209,0.05) 60%,rgba(245, 245, 245,0.05) 60%, rgba(245, 245, 245,0.05) 69%,rgba(164, 164, 164,0.05) 69%, rgba(164, 164, 164,0.05) 100%),linear-gradient(249deg, rgba(248, 248, 248,0.05) 0%, rgba(248, 248, 248,0.05) 32%,rgba(148, 148, 148,0.05) 32%, rgba(148, 148, 148,0.05) 35%,rgba(202, 202, 202,0.05) 35%, rgba(202, 202, 202,0.05) 51%,rgba(181, 181, 181,0.05) 51%, rgba(181, 181, 181,0.05) 100%),linear-gradient(92deg, hsl(214,0%,11%),hsl(214,0%,11%))
}

.Container {
  padding-left: 1%;
  display: flex;
  justify-content: space-evenly;
  align-items: center;

  p {
    width: 45vw;
  }

  .Hello {
    font-family: 'Abril Fatface', cursive;
    font-size: 2em;

    @media (min-width: 10px) and (max-width: 720px){
      font-size: 1.2rem;
    }
  }

  @media (min-width: 10px) and (max-width: 1025px) {
    p {
      width: 75vw;
    }
  }
}

.Headshot {
  max-width: 15%;
  min-width: 200px;
  height: auto;
  margin: 1% 10%;
  border: 1px solid black;
  border-radius: 15px;

  @media (max-width: 1025px) {
    display: none;
  }
}

.CroppedHeadshot {
  display: none;

  @media (max-width: 1025px) {
    display: block;
    border-radius: 50%;
    max-width: 40%;
    height: auto;
  }
}

.Skills {
  padding: 4%;
  padding-top: 0;
  width: 68vw;
  
  span {
    font-family: 'Raleway', sans-serif;
    font-size: .8rem;
    margin-top: 5%;
    margin-bottom: 3%;
  }

  @media (min-width: 10px) and (max-width: 720px){
    span {
      font-size: .7em;
    }
  }

  @media (min-width: 721px) and (max-width: 1025px) {
    span {
      font-size: 1.5em;
    }
  }

  @media (max-width: 1025px){
    padding-left: 0;
    padding-top: 5%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 90vw;
  }
}

.Icons {
  display: flex;
  flex-wrap: wrap;

  p {
    margin: 1%;
    border-radius: 50%;
    height: 11vh;
    width: 11vh;
    background: #ffffff17;
    text-align: center;
    font-family: 'Raleway', sans-serif;
    color: white;
    font-size: .5rem;
    padding: 1%;
    padding-top: 0;
    transition: .5s;
  }

  p:hover {
    background: #0000007d;
    transition: .3s;
  }

  img {
    height: 5vh;
    margin-top: 1em;
  }

  @media (min-width: 300px) and (max-width: 370px){
    p {
      font-size: .3em;
    }
  }

  @media (min-width: 371px) and (max-width: 720px){
    p {
      font-size: .4em;
    }
  }

  @media (min-width: 721px) and (max-width: 1025px){
    p {
      font-size: .5em;
    }
  }

  @media (min-width: 10px) and (max-width: 720px){
    p {
      height: 10vh;
      width: 10vh;
    }

    img {
      height: 4vh;
    }
  }

  @media (max-width: 1025px) {
    justify-content: center;
    width: 90vw;
  }
}

.Resume {
  a {
    font-family: 'Raleway', sans-serif;
    border: 1px solid;
    padding: 2%;
    background: #ffffff1c;
    color: white;
    position: relative;
    left: 70%;
    top: 12vh;
    border-radius: 5px;
    transition: .3s;
    font-size: .8em;
  }
  
  a:hover {
    background: #0000007d;
    transition: .3s;
  }

  @media (max-width: 1025px){
    width: 50%;
    text-align: center;
    padding-top: 5vh;

    a {
      position: unset;
    }
  }
}