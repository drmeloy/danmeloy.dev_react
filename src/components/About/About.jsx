import React from 'react';
import styles from './About.css';
import { Element } from 'react-scroll';

export default function About(){
  return (
    <Element id='About' name='About'>
      <section className={styles.About}>
        <img className={styles.CroppedHeadshot} src='/public/assets/headshot.jfif' alt='headshot of dan' />
        <div className={styles.Container}>
          <p><span className={styles.Hello}>Hello! My name is Dan.</span><br/>I&apos;m a full-stack software engineer. I use technical skill and interpersonal connection to create beautiful and innovative digital products. I believe that meaningful relationships lead to positive results. As such, I seek to invest in people as much as in the technologies we use.</p>
          <img className={styles.Headshot} src='/public/assets/dan.jpg' alt='headshot of dan' />
        </div>
        <div className={styles.Resume}>
          <a href='/public/assets/Dan Meloy Software Engineer Resume.pdf' target='_blank'>View my resume here</a>
        </div>
        <div className={styles.Skills}>
          <span>Technologies I currently use:</span>
          <div className={styles.Icons}>
            <p>
              <img src='/public/assets/js.png' alt='javascript icon'/>
              <br/>Javascript
            </p>
            <p>
              <img src='/public/assets/html.png' alt='html icon'/>
              <br/>HTML
            </p>
            <p>
              <img src='/public/assets/css.png' alt='css icon'/>
              <br/>CSS
            </p>
            <p>
              <img src='/public/assets/node.png' alt='node icon'/>
              <br/>Node.js
            </p>
            <p>
              <img src='/public/assets/react.png' alt='react icon'/>
              <br/>React
            </p>
            <p>
              <img src='/public/assets/redux.png' alt='redux icon'/>
              <br/>Redux
            </p>
            <p>
              <img src='/public/assets/mongodb.png' alt='mongodb icon'/>
              <br/>MongoDB
            </p>
            <p>
              <img src='/public/assets/postgres.png' alt='postgres icon'/>
              <br/>PostgreSQL
            </p>
            <p>
              <img src='/public/assets/github.png' alt='github icon'/>
              <br/>Github
            </p>
            <p>
              <img src='/public/assets/heroku.png' alt='heroku icon'/>
              <br/>Heroku
            </p>
            <p>
              <img src='/public/assets/netlify.png' alt='netlify icon'/>
              <br/>Netlify
            </p>
            <p>
              <img src='/public/assets/jest.png' alt='jest icon'/>
              <br/>Jest
            </p>
            <p>
              <img src='/public/assets/shell.png' alt='shell icon'/>
              <br/>Shell
            </p>
          </div>
          <br/>
          <span>Technologies I am currently learning:</span>
          <div className={styles.Icons}>
            <p>
              <img src='/public/assets/scala.png' alt='scala icon'/>
              <br/>Scala
            </p>
            <p>
              <img src='/public/assets/graphql.png' alt='graphql icon'/>
              <br/>GraphQL
            </p>
            <p>
              <img src='/public/assets/typescript.png' alt='typescript icon'/>
              <br/>Typescript
            </p>
            <p>
              <img src='/public/assets/sass.png' alt='sass icon'/>
              <br/>Sass
            </p>
          </div>
        </div>

      </section>
    </Element>
  );
}
