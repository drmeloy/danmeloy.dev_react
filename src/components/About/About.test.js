import React from 'react';
import { shallow } from 'enzyme';
import About from './About';

describe('About component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<About />);
    expect(wrapper).toMatchSnapshot();
  });
});
