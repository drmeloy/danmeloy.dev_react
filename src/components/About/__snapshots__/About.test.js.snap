// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot 1`] = `
<Element
  id="About"
  name="About"
>
  <section
    className="About"
  >
    <img
      alt="headshot of dan"
      className="CroppedHeadshot"
      src="/public/assets/headshot.jfif"
    />
    <div
      className="Container"
    >
      <p>
        <span
          className="Hello"
        >
          Hello! My name is Dan.
        </span>
        <br />
        I'm a full-stack software engineer. I use technical skill and interpersonal connection to create beautiful and innovative digital products. I believe that meaningful relationships lead to positive results. As such, I seek to invest in people as much as in the technologies we use.
      </p>
      <img
        alt="headshot of dan"
        className="Headshot"
        src="/public/assets/dan.jpg"
      />
    </div>
    <div
      className="Resume"
    >
      <a
        href="/public/assets/Dan Meloy Software Engineer Resume.pdf"
        target="_blank"
      >
        View my resume here
      </a>
    </div>
    <div
      className="Skills"
    >
      <span>
        Technologies I currently use:
      </span>
      <div
        className="Icons"
      >
        <p>
          <img
            alt="javascript icon"
            src="/public/assets/js.png"
          />
          <br />
          Javascript
        </p>
        <p>
          <img
            alt="html icon"
            src="/public/assets/html.png"
          />
          <br />
          HTML
        </p>
        <p>
          <img
            alt="css icon"
            src="/public/assets/css.png"
          />
          <br />
          CSS
        </p>
        <p>
          <img
            alt="node icon"
            src="/public/assets/node.png"
          />
          <br />
          Node.js
        </p>
        <p>
          <img
            alt="react icon"
            src="/public/assets/react.png"
          />
          <br />
          React
        </p>
        <p>
          <img
            alt="redux icon"
            src="/public/assets/redux.png"
          />
          <br />
          Redux
        </p>
        <p>
          <img
            alt="mongodb icon"
            src="/public/assets/mongodb.png"
          />
          <br />
          MongoDB
        </p>
        <p>
          <img
            alt="postgres icon"
            src="/public/assets/postgres.png"
          />
          <br />
          PostgreSQL
        </p>
        <p>
          <img
            alt="github icon"
            src="/public/assets/github.png"
          />
          <br />
          Github
        </p>
        <p>
          <img
            alt="heroku icon"
            src="/public/assets/heroku.png"
          />
          <br />
          Heroku
        </p>
        <p>
          <img
            alt="netlify icon"
            src="/public/assets/netlify.png"
          />
          <br />
          Netlify
        </p>
        <p>
          <img
            alt="jest icon"
            src="/public/assets/jest.png"
          />
          <br />
          Jest
        </p>
        <p>
          <img
            alt="shell icon"
            src="/public/assets/shell.png"
          />
          <br />
          Shell
        </p>
      </div>
      <br />
      <span>
        Technologies I am currently learning:
      </span>
      <div
        className="Icons"
      >
        <p>
          <img
            alt="scala icon"
            src="/public/assets/scala.png"
          />
          <br />
          Scala
        </p>
        <p>
          <img
            alt="graphql icon"
            src="/public/assets/graphql.png"
          />
          <br />
          GraphQL
        </p>
        <p>
          <img
            alt="typescript icon"
            src="/public/assets/typescript.png"
          />
          <br />
          Typescript
        </p>
        <p>
          <img
            alt="sass icon"
            src="/public/assets/sass.png"
          />
          <br />
          Sass
        </p>
      </div>
    </div>
  </section>
</Element>
`;
